### ISSUE TITLE
iRonCub3 (iCubGenova09): BAT board damaged

### ISSUE THREAD

**Issue by gabrielenava** [robotology/icub-tech-support#1594]
### Device name 🤖

iCubGenova09 - iRonCub3

### Request/Failure description

the BAT board got damaged while we were testing the robot for closing another issue: #1588. now, it provides current to the boards even when the motors should be off (the motors power button is not pressed).

### Detailed context

after #1588 we were requested to start the robot and check if the knee calibrates correctly. We did it several times (abour 3-4 times). at a certain point, we heard a spark and see smoke coming out from the right upper leg. We switched everything off and called for support.

After, we powered the motors again and we noted that the boards lights were on even if we did not press the motors button.

**Comment by MrAndrea** [comment]
After an inspection with @Gandoo and @Fabrizio69 I think that some "flying" screws inside the robot made a short circuit on the BAT board, in particular on the HSM circuit of the board. Thus, first I have to check the 4 MOSFETs Q2, Q3, Q4, Q5 and see if they are burned and in case try a replacement since there is no more BAT board available in the warehouse.

**Comment by gabrielenava** [comment]
>After an inspection with @Gandoo and @Fabrizio69 I think that some "flying" screws inside the robot made a short circuit on the BAT board, in particular on the HSM circuit of the board [*]. Thus, first I have to check the 4 MOSFETs Q2, Q3, Q4, Q5 and see if they are burned and in case try a replacement since there is no more BAT board available in the warehouse

indeed when we started the robot the first time a couple of screws fell from the robot on the ground. We were not be able to determine from where the screws fell down, because they were quite big and different from the ones usually used for covers and similar. Also, they seem different from the ones used in the jetpack support (and the ones of the jetpack were all mounted).

however the spark occurred on the right leg upper part. With @Gandoo we unmounted the boards in the upper leg and there were no flying screws there.

Also, we did not see any other screw falling from the robot besides the two that fell on the ground

**Comment by MrAndrea** [comment]
I found mosfet Q4 burned, it was the only one. We have to replace it, and also Q5 since unfortunately during desoldering one pin to check it the package broke.

To manage a quick BAT replacement on the robot we can use a BAT board we took for team Five test (reference: https://github.com/icub-tech-iit/tickets/issues/2824). Then replace the 2 mosfets (iitcode 10127) on the BAT coming from iRonCub3.

cc @Fabrizio69
